src/io/devmgr.c

```diff
@@ -0,0 +1,129 @@
+#include "PR/os_internal.h"
+#include "PR/rcp.h"
+#include "piint.h"
+
+struct test
+{
+    u32 test;
+};
+
+// Is this cheating? Too bad!
+#define LOAD_PHYSICAL_ADDR(out, addr)                    \
+    __asm__ __volatile__("lui %0, %1"                    \
+                         : "=r"(out)                     \
+                         : "i"(PHYS_TO_K1(addr) >> 16)); \
+    __asm__ __volatile__("ori %0, %1, %2"                \
+                         : "=r"(out)                     \
+                         : "r"(out), "i"(PHYS_TO_K1(addr) & 0xFFFF));
```

Comment:
```suggestion
struct test
{
    u32 test;
};

// Is this cheating? Too bad!
#define LOAD_PHYSICAL_ADDR(out, addr)                    \
    __asm__ __volatile__("lui %0, %1"                    \
                         : "=r"(out)                     \
                         : "i"(PHYS_TO_K1(addr) >> 16)); \
    __asm__ __volatile__("ori %0, %1, %2"                \
                         : "=r"(out)                     \
                         : "r"(out), "i"(PHYS_TO_K1(addr) & 0xFFFF));
```
Is any of this still needed?

Reply:
Oops, left that in from how I had it previously matched. Not needed.